pkg/cmd/util/clientcmd/factory.go

```diff
@@ -353,6 +348,114 @@ func (f *Factory) PodForResource(resource string, timeout time.Duration) (string
 	}
 }
 
+func (f *Factory) MostAccuratePodTemplateForObject(object runtime.Object) (*api.PodTemplateSpec, error) {
+	sortBy := func(pods []*v1.Pod) sort.Interface { return controller.ActivePods(pods) }
+	clientset, err := f.ClientAccessFactory.ClientSet()
+	if err != nil {
+		return nil, err
+	}
+	var podSpec *api.PodTemplateSpec
+	var pod *api.Pod
+	var selector labels.Selector
+	var namespace string
+	switch t := object.(type) {
+	case *extensions.ReplicaSet:
+		namespace = t.Namespace
+		selector = labels.SelectorFromSet(t.Spec.Selector.MatchLabels)
+		if err != nil {
+			return &t.Spec.Template, fmt.Errorf("invalid label selector: %v", err)
+		}
+		pods, err := GetAllPods(clientset.Core(), namespace, selector, sortBy)
```

Comment:
You're copy&pasting this piece of code. Move it to outside for the case body, similarly to how upstream does `AttachablePodForObject`.